lua/nvim-treesitter/indent.lua

```diff
@@ -251,7 +282,14 @@ function M.get_indent(lnum)
             if c_is_last_in_line then
               -- If current line is outside the range of a node marked with `@aligned_indent`
               -- Then its indent level shouldn't be affected by `@aligned_indent` node
-              if c_srow and c_srow < lnum - 1 then
+              if c_srow and (c_srow < lnum - 1) then
+                indent = math.max(indent - indent_size, 0)
+              end
+            end
+            if c_is_first_in_line then
+              -- If current line is outside the range of a node marked with `@aligned_indent`
+              -- Then its indent level shouldn't be affected by `@aligned_indent` node
+              if c_srow and (c_srow <= lnum - 1 and metadata["indent.dedent_hanging_closing"]) then
                 indent = math.max(indent - indent_size, 0)
```

Comment:
This one too, but im not sure. WIth your changes it seems like the behavior for this case is different for some cases out there. Of course it won't affect anyone and anything because it's not widely used yet, but may also need to reconsider

Reply:
Second look and if the line contains a single ending delimiter, it may get dedented twice

Reply:
agreed on care needed - the other changes removed some ```@indent.dedents``` which in python were hiding the inherent operation of the ```@align.indent```

Which is actually

```
b = [
    1,
    2,
    3,
    ]
````

if as in python

```
b = [
    1,
    2,
    3,
]
```

is desirable, the new option does that.  The ```@indent.branch``` did that before but also made this complex, and ```@indent.end``` didn't actually do that.

I think you are correct, but we already indented by one just above.  So the indent + dedent gets us to 0, and the next catches the dedent.

However, it reads oddly, should be revisited. 

TBH, I'm pretty surprised by some of the python indents - particularly in the nested collections.  However, PEP8 checkers and other linters for python say it's correct and minor changes report it's no longer correct.

I'll revisit this.